qubes/vm/mix/net.py

```diff
@@ -383,12 +429,48 @@ def detach_network(self):
         if not self.is_running():
             raise qubes.exc.QubesVMNotRunningError(self)
         if self.netvm is None:
-            raise qubes.exc.QubesVMError(
-                self, "netvm should not be {}".format(self.netvm)
+            deferred_from = self.features.get("deferred-netvm", None)
+            if deferred_from is not None:
+                raise qubes.exc.QubesVMError(
+                    self, "netvm should not be {}".format(self.netvm)
+                )
+
+        if self.is_paused():
+            self.log.warning(
+                "Deferred detaching libvirt net device because qube is paused"
             )
+            return
+
+        # Properties extracted from libvirt_domain to support deferred netvm.
+        root = lxml.etree.fromstring(self.libvirt_domain.XMLDesc())
+        interface = root.find(".//interface[@type='ethernet']")
+        if interface is None:
+            return
 
+        class NetVM:  # pylint: disable=too-few-public-methods
+            name = None
+
+        class VM:  # pylint: disable=too-few-public-methods
+            mac = ip = ip6 = None
+            netvm = NetVM()
+
+        vm = VM()
+        backend_elem = interface.find("backenddomain")
+        if backend_elem is not None:
+            netvm_name = backend_elem.get("name")
+            if netvm_name not in self.app.domains:
+                return
+            vm.netvm.name = netvm_name
+        mac_elem = interface.find("mac")
+        if mac_elem is not None:
+            vm.mac = mac_elem.get("address")
+        for ip_elem in interface.findall("ip"):
+            if ip_elem.get("family") == "ipv4":
+                vm.ip = ip_elem.get("address")
+            elif ip_elem.get("family") == "ipv6":
+                vm.ip6 = ip_elem.get("address")
         self.libvirt_domain.detachDevice(
-            self.app.env.get_template("libvirt/devices/net.xml").render(vm=self)
+            self.app.env.get_template("libvirt/devices/net.xml").render(vm=vm)
```

Comment:
You could simply use XML you already got (`interface` variable), instead of re-building it.